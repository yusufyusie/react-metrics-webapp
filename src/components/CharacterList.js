import { React, useEffect, useState } from 'react';
import banner from '../assets/banner.png';
import searchIcon from '../assets/find.svg';
import '../styles/CharacterList.css';
import { useDispatch, useSelector } from 'react-redux';
import { getAllCharacters } from '../redux/characterSlice'
import { Link } from 'react-router-dom';

const CharacterList= () => {
    const { charactersData, loading, error } = useSelector((store) => store.charactersData);
    const dispatch = useDispatch();
    const [search, setSearch] = useState('');

    useEffect(() => {
        dispatch(getAllCharacters());
      }, [dispatch]);

    const handleChange = (e) => {
        e.preventDefault();
        setSearch(e.target.value);
      };

      const filtered = charactersData.filter((person) => person.name.toLowerCase().includes(search));
      if (loading) {
        return (
          <p> Missions are loading!</p>
        );
      }
      if (error) {
        return (
          <p>Something is incorrect!</p>
        );
      }

    return (
    <><section className="character-container">
            <img src={banner} alt="banner" />
            <article className="search-field">
                <img src={searchIcon} alt="search icon" />
                <input
                    type="search"
                    placeholder="Search Characters"
                    onChange={handleChange}
                    value={search} />
            </article>
        </section><section className="character-list">
                {filtered.map((character) => (
                    <Link to={`/details/${character.id}`} key={character.id}></Link>
                ))}
            </section></>
     );
}

export default CharacterList;