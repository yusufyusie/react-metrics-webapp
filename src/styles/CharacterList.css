.character-container {
    width: 100%;
    display: flex;
    flex-direction: column;
    gap: 30px;
    padding: 1rem;
  }
  
  .character-container > img {
    width: 100%;
    height: 225px;
  }
  
  .search-field {
    display: flex;
    align-items: center;
    gap: 8px;
    justify-content: space-between;
    padding: 12px 8px;
    border-radius: 5px;
    background-color: white;
  }
  
  .search-field input {
    font-size: 1rem;
    border: none;
    outline: none;
    width: 100%;
  }
  
  .search-field img {
    width: 17px;
    height: 17px;
  }
  
  .character-list {
    display: grid;
    grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
    gap: 8px;
    margin: 1rem;
  }

  .actorCard {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 15px;
    border-radius: 10px;
    background-color: #2e5984;
  }
  
  .actorCard:hover {
    opacity: 0.8;
  }
  
  .actorImage {
    flex: 2;
    height: 183px;
    width: 275px;
    overflow: hidden;
    display: flex;
    align-items: flex-start;
    cursor: pointer;
  }
  
  .actorImage img {
    min-height: 100%;
    width: 100%;
  }
  
  .actorDesc {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 5px;
  }
  
  .actorDesc h2 {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    max-width: 160px;
    font-size: 1.05rem;
  }
  
  .actorDesc p {
    font-size: 0.8rem;
  }