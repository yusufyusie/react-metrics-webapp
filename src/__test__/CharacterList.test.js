import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import CharacterList from '../components/CharacterList';
import '@testing-library/jest-dom';

describe('CharacterList component test', () => {
  const Varys = {
    fullName: 'Varys',
    id: 19,
  };

  it('Check if the component changed', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <CharacterList
              key={Varys.id}
              id={Varys.id}
              fullName={Varys.fullName}
            />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
