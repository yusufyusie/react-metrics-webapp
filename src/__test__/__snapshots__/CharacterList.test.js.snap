// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component test Check if the component changed 1`] = `
<section
  className="character-container"
>
  <img
    alt="banner"
    src="banner.png"
  />
  <article
    className="search-field"
  >
    <img
      alt="search icon"
      src="find.svg"
    />
    <input
      onChange={[Function]}
      placeholder="Search Characters"
      type="search"
      value=""
    />
  </article>
  <article
    className="character-list"
  />
</section>
`;
